test/functional/PathContainerTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Dhii\Container\FuncTest;
+
+use Dhii\Container\PathContainer as TestSubject;
+use Dhii\Container\TestHelpers\ComponentMockeryTrait;
+use Exception;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use function uniqid;
+
+class PathContainerTest extends TestCase
+{
+    use ComponentMockeryTrait;
+
+    /**
+     * Creates a new instance of the test subject.
+     *
+     * @param array      $dependencies A list of constructor args.
+     * @param array|null $methods      The names of methods to mock in the subject.
+     *
+     * @return MockObject|TestSubject The new instance.
+     * @throws Exception If problem creating.
+     */
+    protected function createSubject(array $dependencies = [], array $methods = null)
+    {
+        return $this->createMockBuilder(TestSubject::class, $methods, $dependencies)
+                    ->getMock();
+    }
+
+    /**
+     * Tests that the subject correctly returns intermediate containers when fetching non-existent keys.
+     *
+     * @since [*next-version*]
+     *
+     * @throws Exception
+     */
+    public function testGet()
+    {
+        $key = 'foo/bar/lorem/ipsum';
+        $value = uniqid('value');
+        $inner = $this->createContainer([
+            $key => $value,
+        ]);
+
+        $delimiter = '/';
+        $subject = $this->createSubject([$inner, $delimiter]);
+
+        $result1 = $subject->get('foo');
+        $this->assertInstanceOf('Psr\Container\ContainerInterface', $result1, 'First result in not a container');
```

Comment:
Didn't quite understand what this means:

> First result in not a container

Reply:
The "first result" is the result of the first `get()` call in the test, since this test fetches multiple values from the container to test the `get()` method. And "not a container" mirrors the assertion: it's not a container instance.